docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+```
+
+**Output**
+```
+MEUCIQC8Zd3NihB/TveLmhk/BAlvuIjSo76fnsywvyyh3JudOgIgbujXo5cMSyBa5i/I9vM3h0HWb/3Waop56hPSRB4Jw5k=
+```
+
+**Submit**
+
+```json
+curl \
+-H "Content-Type: application/json" \
+-H "TxTimeout: 2s" \
+-X POST http://127.0.0.1:6001/data/tx \
+--data '{
+  "payload": {
+    "must_sign_user_ids": [
+      "alice"
+    ],
+    "tx_id": "Tx001",
+    "db_operations": [
+      {
+        "db_name": "db2",
+        "data_reads": [
+          {
+            "key": "key1",
+            "version": {
+              "block_num": 5
+            }
+          },
+          {
+            "key": "key2",
+            "version": {
+              "block_num": 5
+            }
+          }
+        ],
+        "data_writes": [
+          {
+            "key": "key2",
+            "value": "dGhpcyBpcyBhIHNlY29uZCB2YWx1ZSB1cGRhdGVk",
+            "acl": {
+              "read_users": {
+                "alice": true
+              },
+              "read_write_users": {
+                "alice": true
+              }
+            }
+          },
+          {
+            "key": "key1",
+            "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVlIHVwZGF0ZWQ=",
+            "acl": {
+              "read_users": {
+                "alice": true
+              },
+              "read_write_users": {
+                "alice": true
+              }
+            }
+          }
+        ],
+        "data_deletes": [
+          {
+            "key": "key3"
+          }
+        ]
+      }
+    ]
+  },
+  "signatures": {
+    "alice": "MEUCIQC8Zd3NihB/TveLmhk/BAlvuIjSo76fnsywvyyh3JudOgIgbujXo5cMSyBa5i/I9vM3h0HWb/3Waop56hPSRB4Jw5k="
+  }
+}' | jq . 
+
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "receipt": {
+      "header": {
+        "base_header": {
+          "number": 6,
+          "previous_base_header_hash": "TJYT7zr95D3ghtU/M+j8OStU6r8Y+XRC5xfb90jshbI=",
+          "last_committed_block_hash": "r57EtiB2XO1XlVAxBwFtn7cHP7YsfUBFi69y0UY0tw0=",
+          "last_committed_block_num": 5
+        },
+        "skipchain_hashes": [
+          "r57EtiB2XO1XlVAxBwFtn7cHP7YsfUBFi69y0UY0tw0="
+        ],
+        "tx_merkel_tree_root_hash": "dSc42JVp36bfORFO8QehzQFypEtBQfUzluT6ixhjeo4=",
+        "state_merkel_tree_root_hash": "YRsFw5PcG1XzYO5o5nJa3lVfknijRqu6cCaDF1zpPA8=",
+        "validation_info": [
+          {}
+        ]
+      }
+    }
+  },
+  "signature": "MEUCIHGNm7923sl7r3o/UBPi6IA4LhmV7vjISxB2aRTIJI8uAiEA4EKMhVJIwKMv1Nzy7zDaftEkal2wFl3pQN9heKw6tCY="
+}
+```
```

Comment:
After the preparation, or just before, it will be useful to add a table (or tables) of what the database state is per block. i.e:

at block X:
key1 / Value1 / Version
key2 / Value2 / Version

etc